With a little help from my friend, the native-cctools wrapping ld to
support '-soname' flag on AIX, we can provide full soname support there
(#213277).  However, this patch is not for upstream, as they cannot rely
on that ld wrapper: They would have to do the aix-soname magic themself
instead.

--- configure
+++ configure
@@ -125,6 +125,7 @@
   case "$uname" in
   Linux* | linux* | GNU | GNU/* | *BSD | *bsd* | DragonFly | solaris*)
         LDSHARED=${LDSHARED-"$cc -shared -Wl,-soname,libz.so.1,--version-script,zlib.map"} ;;
+  AIX* | aix* ) LDSHARED=${LDSHARED-"$cc -shared -Wl,-soname,libz.so.1"} ;;
   CYGWIN* | Cygwin* | cygwin* | OS/2*)
         EXE='.exe' ;;
   MINGW* | mingw*)
